src/KineticsStripOutputNode.cpp

```diff
@@ -0,0 +1,102 @@
+#include "KineticsStripOutputNode.h"
+#include <iostream>
+#include <algorithm>
+
+// Reusable state for sending data packets
+static int out_fd = -1;
+// static struct sockaddr_in out_addr;
+
+// This is dynamic because we need to allocate memory based on
+// the number of pixels per strip.
+unsigned char out_packet[512] = {};
+
+std::string ip;
+int startport, width, height;
+struct sockaddr_in out_addr;
+
+KineticsStripOutputNode::KineticsStripOutputNode(Context *context, QSize chainSize)
+: SelfTimedReadBackOutputNode(context, chainSize, 10) {
+
+  connect(this, &SelfTimedReadBackOutputNode::frame, this, &KineticsStripOutputNode::onFrame, Qt::DirectConnection);
+
+  // Try to open a socket
+  if (out_fd > 0) {
+    close(out_fd);
+  }
+  out_fd = socket(AF_INET, SOCK_DGRAM, 0);
+  if (out_fd < 0) {
+    qDebug() << "Error opening Kinet data socket";
+  }
+
+  /*
+  data packet header (
+  ("magic", "I", 0x0401dc4a),
+  ("version", "H", 0x0100),
+  ("type", "H", 0x0101),
+  ("sequence", "I", 0x00000000),
+  ("port", "B", 0x00),
+  ("padding", "B", 0x00),
+  ("flags", "H", 0x0000),
+  ("timer", "I", 0xffffffff),
+  ("universe", "B", 0x00),)*/
+
+  // 21 bytes for the header; 3 bytes (RGB) for each pixel
+  //magic, version, type, sequence, port, padding, flags, timer, universe
+  unsigned char header[21] = {0x04, 0x01, 0xdc, 0x4a, 0x01, 0x00, 0x01, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0xff, 0xff, 0xff, 0x00};
+  std::copy(header, header + 21, out_packet);
```

Comment:
I would prefer if you used Qt's containers here (probably they are more compatible with `QUdpSocket` as well)